scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1516,6 +1536,152 @@
                    max-considerable)
                  next-considerable))))))
 
+(defn job->task-txn
+  "Generate task transaction for a job."
+  [job metadata compute-cluster]
+  (let [{:keys [job/last-waiting-start-time job/uuid]} job
+        {:keys [executor task-id]} metadata
+        job-ref [:job/uuid uuid]
+        instance-start-time (now)]
+    [[:job/allowed-to-start? job-ref]
+     ; The job will remain waiting until the instance is running.
+     [:db/add job-ref :job/state :job.state/waiting]
+     (cond->
```

Comment:
Overall, this code feels like a copy&paste from `matches->task-txns` Can we use that function, or refactor a common helper function that this and `matches->task-txns` uses?

Reply:
Yes, there's quite a bit of overlap here in this and other parts to get it to work. Happy to discuss refactoring with you tomorrow? I want us to consider how much is useful to do now vs after the production test.